guide/popular-topics/canvas.md

```diff
@@ -390,164 +268,48 @@ The purpose of this small section is to demonstrate that working with Canvas is
 
 Since we covered how to load external images and fix dimensions, let's turn the avatar into a circle to improve the image's overall style.
 
-<branch version="11.x">
-
-```js
-client.on('guildMemberAdd', async member => {
-	const channel = member.guild.channels.find(ch => ch.name === 'member-log');
-	if (!channel) return;
-
-	const canvas = Canvas.createCanvas(700, 250);
-	const ctx = canvas.getContext('2d');
-
-	const background = await Canvas.loadImage('./wallpaper.jpg');
-	ctx.drawImage(background, 0, 0, canvas.width, canvas.height);
-
-	ctx.strokeStyle = '#74037b';
-	ctx.strokeRect(0, 0, canvas.width, canvas.height);
-
-	// Pick up the pen
-	ctx.beginPath();
-	// Start the arc to form a circle
-	ctx.arc(125, 125, 100, 0, Math.PI * 2, true);
-	// Put the pen down
-	ctx.closePath();
-	// Clip off the region you drew on
-	ctx.clip();
-
-	const avatar = await Canvas.loadImage(member.user.displayAvatarURL);
-	ctx.drawImage(avatar, 25, 25, 200, 200);
-
-	const attachment = new Discord.Attachment(canvas.toBuffer(), 'welcome-image.png');
-
-	channel.send(`Welcome to the server, ${member}!`, attachment);
-});
-```
-
-</branch>
-<branch version="12.x">
-
-```js
+```js {5-12}
 client.on('guildMemberAdd', async member => {
-	const channel = member.guild.channels.cache.find(ch => ch.name === 'member-log');
-	if (!channel) return;
-
-	const canvas = Canvas.createCanvas(700, 250);
-	const ctx = canvas.getContext('2d');
-
-	const background = await Canvas.loadImage('./wallpaper.jpg');
-	ctx.drawImage(background, 0, 0, canvas.width, canvas.height);
-
-	ctx.strokeStyle = '#74037b';
-	ctx.strokeRect(0, 0, canvas.width, canvas.height);
+	// ...
+	context.strokeRect(0, 0, canvas.width, canvas.height);
 
 	// Pick up the pen
-	ctx.beginPath();
+	context.beginPath();
 	// Start the arc to form a circle
-	ctx.arc(125, 125, 100, 0, Math.PI * 2, true);
+	context.arc(125, 125, 100, 0, Math.PI * 2, true);
 	// Put the pen down
-	ctx.closePath();
+	context.closePath();
 	// Clip off the region you drew on
-	ctx.clip();
-
-	const avatar = await Canvas.loadImage(member.user.displayAvatarURL({ format: 'jpg' }));
-	ctx.drawImage(avatar, 25, 25, 200, 200);
-
-	const attachment = new Discord.MessageAttachment(canvas.toBuffer(), 'welcome-image.png');
-
-	channel.send(`Welcome to the server, ${member}!`, attachment);
+	context.clip();
+	// ...
```

Comment:
Doesn't this completely remove all mentions of the avatar this part is supposed to make into a circle? I know that this way `<branch>`ing can get removed, but i think that in this case it's needed here,

Reply:
It doesn't remove it; the `strokeRect()` line is added one codeblock before the avatar `load/drawImage()` lines. The clip code should go after `strokeRect()` and are mentioned one codeblock after the avatar code.

For better context: https://deploy-preview-644--discordjs-guide.netlify.app/popular-topics/canvas.html#manipulating-images

Reply:
Don't they have completely different parameters? (the one removed here, and one left before)